frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/BrowseNavSection.unit.spec.tsx

```diff
@@ -0,0 +1,119 @@
+    jest.restoreAllMocks();
+  });
+
+  it("should render Models, Databases, and Metrics when not embedding in iframe", () => {
+    setup();
+
+    expect(
+      screen.getByRole("listitem", { name: "Browse models" }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole("listitem", { name: "Browse databases" }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole("listitem", { name: "Browse metrics" }),
+    ).toBeInTheDocument();
+  });
+
+  describe("interactive embedding with `entity_types` (EMB-229)", () => {
+    it("should show models and tables when no `entity_types` is provided", () => {
+      setup({ isEmbeddingIframe: true });
+
+      expect(
+        screen.getByRole("listitem", { name: "Browse models" }),
+      ).toBeInTheDocument();
+      expect(
+        screen.getByRole("listitem", { name: "Browse databases" }),
+      ).toBeInTheDocument();
+      expect(
+        screen.queryByRole("listitem", { name: "Browse metrics" }),
+      ).not.toBeInTheDocument();
+    });
+
+    it('should show models and tables when no `entity_types` is `["model", "table"]`', () => {
+      setup({ isEmbeddingIframe: true, entityTypes: ["model", "table"] });
+
+      expect(
+        screen.getByRole("listitem", { name: "Browse models" }),
+      ).toBeInTheDocument();
+      expect(
+        screen.getByRole("listitem", { name: "Browse databases" }),
+      ).toBeInTheDocument();
+      expect(
+        screen.queryByRole("listitem", { name: "Browse metrics" }),
+      ).not.toBeInTheDocument();
+    });
+
+    it('should show only models when no `entity_types` is `["model"]`', () => {
+      setup({ isEmbeddingIframe: true, entityTypes: ["model"] });
+
+      expect(
+        screen.getByRole("listitem", { name: "Browse models" }),
+      ).toBeInTheDocument();
+      expect(
+        screen.queryByRole("listitem", { name: "Browse databases" }),
+      ).not.toBeInTheDocument();
+      expect(
+        screen.queryByRole("listitem", { name: "Browse metrics" }),
+      ).not.toBeInTheDocument();
+    });
+
+    it('should show only tables when no `entity_types` is `["table"]`', () => {
+      setup({ isEmbeddingIframe: true, entityTypes: ["table"] });
+
+      expect(
+        screen.queryByRole("listitem", { name: "Browse models" }),
+      ).not.toBeInTheDocument();
+      expect(
+        screen.getByRole("listitem", { name: "Browse databases" }),
+      ).toBeInTheDocument();
+      expect(
+        screen.queryByRole("listitem", { name: "Browse metrics" }),
+      ).not.toBeInTheDocument();
+    });
+  });
+});
+
+interface SetupOpts {
+  isEmbeddingIframe?: boolean;
+  entityTypes?: EmbeddingEntityType[];
+}
+
+function setup({ isEmbeddingIframe, entityTypes }: SetupOpts = {}) {
+  if (isEmbeddingIframe) {
+    jest.spyOn(domUtils, "isWithinIframe").mockReturnValue(true);
+  }
+
+  renderWithProviders(
+    <BrowseNavSection
+      hasDataAccess
+      nonEntityItem={{ type: "non-entity", url: "/" }}
```

Comment:
this prop is required, and this value is grabbed directly from React devtools. I'm not going to dive into why it has to be this value, or why it's needed.

Reply:
aight, that's a very good point. I would forget this in the future for sure.

Reply:
Fixed in 86cebd61407845911efcb2190a935e0ac47ec0a5